docs/src/stratospheric_reaction.md

```diff
@@ -0,0 +1,288 @@
+
+As mentioned above, we will try two approaches to solve this PDS and compare their efficiency. These are
+1. an out-of-place implementation with standard (dynamic) matrices and vectors,
+2. an out-of-place implementation with static matrices and vectors from [StaticArrays.jl](https://juliaarrays.github.io/StaticArrays.jl/stable/).
+
+### Standard out-of-place implementation
+
+Here we create a function to compute the production matrix with return type `Matrix{Float64}` and a second function for the destruction vector with return type `Vector{Float64}`.
+
+```@example stratreac
+using PositiveIntegrators # load PDSProblem
+
+function prod(u, p, t)
+    O1D, O, O3, O2, NO, NO2 = u
+
+    Tr = 4.5
+    Ts = 19.5
+    T = mod(t / 3600, 24)
+    if (Tr <= T) && (T <= Ts)
+        Tfrac = (2 * T - Tr - Ts) / (Ts - Tr)
+        sigma = 0.5 + 0.5 * cos(pi * abs(Tfrac) * Tfrac)
+    else
+        sigma = 0.0
+    end
+
+    M = 8.120e16
+
+    k1 = 2.643e-10 * sigma^3
+    k2 = 8.018e-17
+    k3 = 6.120e-4 * sigma
+    k4 = 1.567e-15
+    k5 = 1.070e-3 * sigma^2
+    k6 = 7.110e-11
+    k7 = 1.200e-10
+    k8 = 6.062e-15
+    k9 = 1.069e-11
+    k10 = 1.289e-2 * sigma
+    k11 = 1.0e-8
+
+    r1 = k1 * O2
+    r2 = k2 * O * O2
+    r3 = k3 * O3
+    r4 = k4 * O3 * O
+    r5 = k5 * O3
+    r6 = k6 * M * O1D
+    r7 = k7 * O1D * O3
+    r8 = k8 * O3 * NO
+    r9 = k9 * NO2 * O
+    r10 = k10 * NO2
+    r11 = k11 * NO * O
+
+    return [0.0 0.0 r5 0.0 0.0 0.0;
+            r6 r1+r10 r3 r1 0.0 0.0;
+            0.0 r2 0.0 0.0 0.0 0.0;
+            r7 r4+r9 r4+r7+r8 r3+r5 0.0 0.0;
+            0.0 0.0 0.0 0.0 0.0 r9+r10;
+            0.0 0.0 0.0 0.0 r8+r11 0.0]
+end
+
+function dest(u, p, t)
+    O1D, O, O3, O2, NO, NO2 = u
+
+    k2 = 8.018e-17
+    k11 = 1.0e-8
+
+    r2 = k2 * O * O2
+    r11 = k11 * NO * O
+
+    return [0.0, r11, 0.0, r2, 0.0, 0.0]
+end
+nothing #hide
+```
+The solution of the stratospheric reaction problem can now be computed as follows.
+```@example stratreac
+u0 = [9.906e1, 6.624e8, 5.326e11, 1.697e16, 4e6, 1.093e9] # initial values
+tspan = (4.32e4, 3.024e5) # time domain
+prob = PDSProblem(prod, dest, u0, tspan) # create the PDS
+
+sol = solve(prob, MPRK43I(1.0, 0.5))
+
+nothing #hide
+```
+```@example stratreac
+using Plots
+
+plot(sol,
+    layout=(3,2),
+    xguide = "t [h]",
+    xguidefontsize = 8,
+    xticks = (range(first(tspan), last(tspan), 4), range(12.0, 84.0, 4)),
+    yguide=["O¹ᴰ" "O" "O₃" "O₂" "NO" "NO₂"],
+    tickfontsize = 7,
+    legend = :none, 
+    widen = true
+    )
+```
+
+As MPRK schemes do not preserve general linear invariants, especially when applied to non-conservative PDS, we compute and plot the relative errors with respect to both linear invariants to see how well these are preserved.
```

Comment:
Shall we also show some results with methods from OrdinaryDiffEq.jl?

Reply:
I thought about it, but thought this would fit better in the "Benchmarks" section suggested in #7. Do you agree?

Reply:
It would be nice to include this in some benchmarks, too 👍 However, I think it would still be nice to have a qualitative comparison here with one standard method that is able to conserve all linear invariants.

Reply:
Added.